Mathlib/Combinatorics/SimpleGraph/Matching.lean

```diff
@@ -235,7 +237,60 @@ lemma odd_matches_node_outside {u : Set V} {c : ConnectedComponent (Subgraph.del
     Finset.card_image_of_injective _ (Subtype.val_injective), Set.toFinset_card] at hMeven ⊢
   exact hMeven
 
+
 end Finite
+
+lemma isClique_even_iff_matches [DecidableEq V] (u : Set V) (hu : Set.Finite u)
+    (hc : G.IsClique u) : Even (Nat.card u) ↔ ∃ (M : Subgraph G), M.verts = u ∧ M.IsMatching := by
+  haveI : Fintype u := hu.fintype
+  refine ⟨?_ , by
+    rintro ⟨M, ⟨hMl, hMr⟩⟩
+    haveI : Fintype M.verts := hMl ▸ hu.fintype
+    subst hMl
+    simpa [Nat.card_eq_card_finite_toFinset hu, Set.toFinite_toFinset,
+      Set.toFinset_card] using hMr.even_card⟩
+  intro he
+  cases' Set.eq_empty_or_nonempty u with h h
+  · subst h
+    let empty : Subgraph G := {
+      verts := ∅,
+      Adj := fun _ _ ↦ False,
+      adj_sub := False.elim,
+      edge_vert := False.elim }
```

Comment:
Is that not just `⊥`?
```suggestion
    let empty : Subgraph G := ⊥
```
Then you don't even need a `let`

Reply:
Ah, that's right! Fixed:)